### ISSUE TITLE
Change to WSL2 causing failure to start the service

### ISSUE THREAD

**Issue by fruch** [docker/for-win#7197]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 9ABD25F8-BDFD-4206-9A41-DA6FA169A614/20200611210733

### Information
I was working with docker hyper-visor, and got a pop that my windows now support WSL2
and if I want to switch

I've answer yes, and did the switch

### Steps to reproduce the behavior

Now I'm getting this error when booting
I've tried to uninstall and insatll again, didn't helped

```
WSL distro stopped while waiting for Lifecycle server.
  >Stdout:

  >Stderr:
2020/06/11 21:07:31 resolving /mnt/host/c/Program Files/Docker/Docker/resources/wsl/docker-wsl-cli.iso...
2020/06/11 21:07:31 copying...
Error: mounting wslCLIDest: open /mnt/host/wsl/docker-desktop-data/isocache/52d7fd09909e14e718842ad085b334d7345fd6c266fdbcd8036db2d773109b54.iso: input/output error
2020/06/11 21:07:31 open /mnt/host/wsl/docker-desktop-data/isocache/52d7fd09909e14e718842ad085b334d7345fd6c266fdbcd8036db2d773109b54.iso: input/output error
mounting wslCLIDest
main.doRun
	/go/src/github.com/docker/pinata/linuxkit/pkg/wsl-bootstrap/main.go:224
main.run.func1
	/go/src/github.com/docker/pinata/linuxkit/pkg/wsl-bootstrap/main.go:83
github.com/docker/pinata/vendor/github.com/spf13/cobra.(*Command).execute
	/go/src/github.com/docker/pinata/vendor/github.com/spf13/cobra/command.go:762
github.com/docker/pinata/vendor/github.com/spf13/cobra.(*Command).ExecuteC
	/go/src/github.com/docker/pinata/vendor/github.com/spf13/cobra/command.go:852
github.com/docker/pinata/vendor/github.com/spf13/cobra.(*Command).Execute
	/go/src/github.com/docker/pinata/vendor/github.com/spf13/cobra/command.go:800
main.main
	/go/src/github.com/docker/pinata/linuxkit/pkg/wsl-bootstrap/main.go:25
runtime.main
	/usr/local/go/src/runtime/proc.go:203
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1357
```

**Comment by fruch** [comment]
I've found a work around
```
wsl --unregister docker-desktop-data
```
and restart the docker desktop, and it started working....

**Comment by IroNEDR** [comment]
@fruch This solved my problem too. docker-desktop was not starting on windows 10 home.   
When I checked using `wsl -l -v` it showed that the subsystem called 'docker-desktop' was stuck at the state "installing".

Even reinstalling docker did not work. The "docker-desktop" would disappear when I uninstalled but came back up as "installing" in the wsl list after reinstalling docker.

Running `wsl --unregister docker-desktop` solved the problem and docker now starts up fine
